src/stores/configurationStore.ts

```diff
@@ -0,0 +1,151 @@
+  state: (): ConfigurationStateIF => ({
+    configuration: null
+  }),
+
+  getters: {
+    getAuthWebUrl (state): string {
+      return state.configuration.VUE_APP_AUTH_WEB_URL
+    },
+
+    getMyBusinessRegistryUrl (): string {
+      return this.getAuthWebUrl + 'business'
+    },
+
+    getBusinessProfileUrl (): string {
+      return this.getAuthWebUrl + 'businessprofile'
+    },
+
+    getRegHomeUrl (state): string {
+      return state.configuration?.VUE_APP_REGISTRY_HOME_URL || ''
+    },
+
+    getLoginUrl (): string {
+      return this.getRegHomeUrl + 'login'
+    },
+
+    getBusinessUrl (state): string {
+      return state.configuration?.VUE_APP_BUSINESSES_URL || ''
+    },
+
+    getCreateUrl (state): string {
+      return state.configuration.VUE_APP_BUSINESS_CREATE_URL
+    },
+
+    getEditUrl (state): string {
+      return state.configuration.VUE_APP_BUSINESS_EDIT_URL
+    },
+
+    getLegalApiUrl (state): string {
+      const root = state.configuration
+      return root.VUE_APP_LEGAL_API_URL + root.VUE_APP_LEGAL_API_VERSION_2 + '/'
+    },
+
+    getAuthApiUrl (state): string {
+      if (state.configuration?.VUE_APP_AUTH_API_URL && state.configuration?.VUE_APP_AUTH_API_VERSION) {
+        return state.configuration.VUE_APP_AUTH_API_URL + state.configuration.VUE_APP_AUTH_API_VERSION + '/'
+      }
+      return ''
+    },
+
+    getPayApiUrl (state): string {
+      if (state.configuration?.VUE_APP_PAY_API_URL && state.configuration?.VUE_APP_PAY_API_VERSION) {
+        return state.configuration.VUE_APP_PAY_API_URL + state.configuration.VUE_APP_PAY_API_VERSION + '/'
+      }
+      return ''
+    },
+
+    getStatusApiUrl (state): string {
+      return state.configuration.VUE_APP_STATUS_API_URL + state.configuration.VUE_APP_STATUS_API_VERSION
+    },
+
+    getSiteminderLogoutUrl (state): string {
+      return state.configuration.VUE_APP_SITEMINDER_LOGOUT_URL
+    },
+
+    getHotJarId (state): string {
+      return state.configuration.VUE_APP_HOTJAR_ID
+    },
+
+    getAddressCompleteKey (state): string {
+      return state.configuration.VUE_APP_ADDRESS_COMPLETE_KEY
+    },
+
+    /** Get Launch Darkly Client ID */
+    getBusinessFilingLdClientId (state): string {
+      return state.configuration.VUE_APP_BUSINESS_FILING_LD_CLIENT_ID
+    },
+
+    getSentryDsn (state): string {
+      return state.configuration.VUE_APP_SENTRY_DSN
+    }
+  },
+
+  actions: {
+    setConfiguration (data: any) {
+      this.configuration = data
+    },
+
+    /**
+     * Use this mutator to set a specific attribute in unit testing. Eg,
+     * store.commit('setTestConfiguration', { key: 'VUE_APP_PAY_API_URL', value: 'https://auth.web.url/' })
+     */
+    setTestConfiguration (state: ConfigurationStateIF, { key, value }: KeyValueIF) {
+      if (!state.configuration) {
+        Vue.set(state, 'configuration', {})
+      }
+      Vue.set(state.configuration, key, value)
```

Comment:
Looks like this line should be in an "else" block, or we might be setting a falsy (null/undefined) value in Vue.set().

Please fix and test (per existing unit tests).

Reply:
This works, right?